### PR TITLE
RDM-8530 - BEFTA integration

### PR DESCRIPTION
**Pull request by mattnayler**
### JIRA link (if applicable) ###

[RDM-8530](https://tools.hmcts.net/jira/browse/RDM-8530) - _Manage Case Assignment - BEFTA integration_

### Change description ###

* Register BEFTA framework and add initial sample test for health endpoint
* Suppress Java 11 illegal reflective access warnings from BEFTA framework
* Bump PMD version number to remove XPath deprecated attribute warnings.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -9,9 +9,52 @@ def component = "manage-case-assignment"
...
+    secretType : 'Secret',
+    name       : secretName,
+    version    : '',
+    envVariable: envVar
+  ]
+}
+
+def vaultOverrides = [
+  'preview' : 'aat',
+  'spreview': 'saat'
```
Comment (smathangi) [inline Jenkinsfile_CNP:36]: spreview /saat is not available  anymore. Please remove.

File: README.md [hunk 2]
```diff
@@ -48,6 +48,32 @@ To run all integration tests execute the following command:
 ./gradlew integration
 ```
 
+### Functional tests
+These are the tests run against an environment. For example if you would like to test your local
+ environment you'll need to export the following variables on your `.bash_profile` script.
+
+```bash
```
Comment (smathangi) [inline README.md:55]: Thanks 👍

File: build.gradle [hunk 3]
```diff
@@ -212,9 +235,15 @@ dependencies {
   testImplementation group: 'io.rest-assured', name: 'json-path', version: versions.restAssured
   testImplementation group: 'io.rest-assured', name: 'xml-path', version: versions.restAssured
 
+  // groovy v3.* added to avoid 'WARNING: Illegal reflective access by org.codehaus.groovy.reflection.CachedClass'
+  // in groovy:2.5.* when running functional/befta tests
+  testImplementation group: 'org.codehaus.groovy', name: 'groovy', version: '3.0.3'
```
Comment (smathangi) [inline build.gradle:240]: Just for my knowledge - where we use groovy?

Comment (mattnayler) [reply build.gradle:240]: Not directly.

Even without befta (i.e. outside of this PR) the functional tests will generate warnings for groovy-2.5.11.

Adding `jvmArgs "--illegal-access=debug"` to the _functional_ task config generates full stack traces that point at _io.restassured_. If I increase the restAssured version to 4.3.0 (which is supposed to include the switchover to groovy 3.*) we now get only one warning (still referencing groovy 2.5.11) but we now also get an AbstractMethodError.  So decided to just bump groovy as it appears to resolve it.

Was tempted to use the dependencyManagement config to force the groovy version number but as the warnings only occurred during test but I didn't know If using DM would effect the libraries included in the main build.

File: Jenkinsfile_CNP (hunk 1 above)
Comment (mattnayler) [reply Jenkinsfile_CNP:36]: 👍 and I will add a comment to explain why CCD vault is being referenced.

**Pull request merged**